test_scripts/Defects/8_0/3541_Resumption_of_MenuLayout.lua

```diff
@@ -0,0 +1,145 @@
+            }
+          }
+        }
+      }
+    }
+  }
+  onSysCaps.appID = common.getHMIAppId()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnSystemCapabilityUpdated", onSysCaps)
+end
+
+local function sendSetGP()
+  common.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      hashId = data.payload.hashID
+    end)
+  local dataToHMI = utils.cloneTable(sgpParams)
+  dataToHMI.appID = common.getHMIAppId()
+  local cid = common.getMobileSession():SendRPC("SetGlobalProperties", sgpParams)
+  common.getHMIConnection():ExpectRequest("UI.SetGlobalProperties", dataToHMI)
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function sendAddSubMenu()
+  common.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      hashId = data.payload.hashID
+    end)
+  local dataToHMI = {
+    menuID = asmParams.menuID,
+    menuLayout = asmParams.menuLayout,
+    menuParams = {
+      menuName = asmParams.menuName
+    }
+  }
+  dataToHMI.appID = common.getHMIAppId()
+  local cid = common.getMobileSession():SendRPC("AddSubMenu", asmParams)
+  common.getHMIConnection():ExpectRequest("UI.AddSubMenu", dataToHMI)
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function unexpectedDisconnect()
+  common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  common.mobile.disconnect()
+  common.run.wait(1000)
+end
+
+local function reRegisterApp()
+  common.getMobileSession():StartService(7)
+  :Do(function()
+    local appParams = utils.cloneTable(common.app.getParams())
+    appParams.hashID = hashId
+    local cid = common.getMobileSession():SendRPC("RegisterAppInterface", appParams)
+    common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+    :Do(function()
+        local dataToHMIsgp = utils.cloneTable(sgpParams)
+        dataToHMIsgp.appID = common.getHMIAppId()
+        common.getHMIConnection():ExpectRequest("UI.SetGlobalProperties", dataToHMIsgp)
+        :Do(function(_, data)
+            utils.printTable(data)
+            common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+          end)
+        local dataToHMIasm = {
+          menuID = asmParams.menuID,
+          menuLayout = asmParams.menuLayout,
+          menuParams = {
+            menuName = asmParams.menuName
+          }
+        }
+        dataToHMIasm.appID = common.getHMIAppId()
+        common.getHMIConnection():ExpectRequest("UI.AddSubMenu", dataToHMIasm)
+        :Do(function(_, data)
+            utils.printTable(data)
```

Comment:
Do we need a printTable(data) in the test?

Reply:
It's just debug info and can be removed

Reply:
done in cca0c3f